core/Mail.php

```diff
@@ -42,15 +42,21 @@ public function setDefaultFromPiwik()
         /** @var Translator $translator */
         $translator = StaticContainer::get('Piwik\Translation\Translator');
 
-        $fromEmailName = Config::getInstance()->General['noreply_email_name'];
+        $fromEmailName = Config::getInstance()->mail['noreply_email_name'];
+        if(empty($fromEmailName)){
+            $fromEmailName = Config::getInstance()->General['noreply_email_name'];
```

Comment:
I reckon it might be better to create an update script using the command `./console generate:update --component=core` to migrate both config settings from `General` to `mail` within the `doUpdate()` method. This way it will be easier to change the value for users in the future (by directly seeing the default value `noreply@{DOMAIN}`) and it won't be defined in 2 places (less confusion).

You can see an example here: https://github.com/matomo-org/matomo/blob/3.x-dev/core/Updates/2.15.0-b12.php#L25-L41 

This means you can set the default value of `noreply@{DOMAIN}` for `[mail]noreply...`